### ISSUE TITLE
MVCSwagParam ptArray Bug

### ISSUE THREAD

**Issue by st1gok** [danieleteti/delphimvcframework#377]
Hi!
I use RC6 and applied changes from fix https://github.com/danieleteti/delphimvcframework/issues/376

When you use
MVCSwagParam(plQuery, 'types', 'types', ptArray, false,'','money,loading,unloading')

result in json: 
{
                        "in": "query",
                        "name": "types",
                        "description": "types",
                        "type": "array",
                        "enum": [
                            "money",
                            "loading",
                            "unloading"
                        ]
                    }

but it should be:
{
            "name": "types",
            "in": "query",
            "description": "types",
            "type": "array",
            **"items": {**
              **"type": "string",**
              "enum": [
                "money",
                "loading",
                "unloading"
              ]
            }
          }

**Comment by joaoduarte19** [comment]
I will check this behavior

**Comment by st1gok** [comment]
Also I have problem with [MVCSwagResponses(200, 'ok', INDEX_JSON_SCHEMA)] where 
      INDEX_JSON_SCHEMA =
    '{' + sLineBreak +
    **'    "type": "array",' + sLineBreak +
    '    "items": {' + sLineBreak +**
    '    "type": "object",' + sLineBreak +
    '    "properties": {' + sLineBreak +
    '        "application": {' + sLineBreak +
    '            "type": "string",' + sLineBreak +
    '            "description": "Application Name"' + sLineBreak +
    '        },' + sLineBreak +
    '        "online": {' + sLineBreak +
    '            "type": "boolean",' + sLineBreak +
    '            "description": "Defines if the server is online"' + sLineBreak +
    '        },' + sLineBreak +
    '        "serverdatetime": {' + sLineBreak +
    '            "type": "string",' + sLineBreak +
    '            "description": "Current server time"' + sLineBreak +
    '        }' + sLineBreak +
    '    }' + sLineBreak +
    '        ]},' + sLineBreak +
    '}';

maybe these errors are related

**Comment by joaoduarte19** [comment]
> Hi!
> I use RC6 and applied changes from fix #376
> 
> When you use
> MVCSwagParam(plQuery, 'types', 'types', ptArray, false,'','money,loading,unloading')
> 
> result in json:
> {
> "in": "query",
> "name": "types",
> "description": "types",
> "type": "array",
> "enum": [
> "money",
> "loading",
> "unloading"
> ]
> }
> 
> but it should be:
> {
> "name": "types",
> "in": "query",
> "description": "types",
> "type": "array",
> **"items": {**
> **"type": "string",**
> "enum": [
> "money",
> "loading",
> "unloading"
> ]
> }
> }

Using the last commit from dmvc I did these tests and when I define the parameter type as an array, the generated JsonSchema shows it as an array. However when you use enumerators the value of an enumerator cannot be an array. See https://swagger.io/docs/specification/2-0/enums/

**Comment by joaoduarte19** [comment]
> Also I have problem with [MVCSwagResponses(200, 'ok', INDEX_JSON_SCHEMA)] where
> INDEX_JSON_SCHEMA =
> '{' + sLineBreak +
> **' "type": "array",' + sLineBreak + ' "items": {' + sLineBreak +**
> ' "type": "object",' + sLineBreak +
> ' "properties": {' + sLineBreak +
> ' "application": {' + sLineBreak +
> ' "type": "string",' + sLineBreak +
> ' "description": "Application Name"' + sLineBreak +
> ' },' + sLineBreak +
> ' "online": {' + sLineBreak +
> ' "type": "boolean",' + sLineBreak +
> ' "description": "Defines if the server is online"' + sLineBreak +
> ' },' + sLineBreak +
> ' "serverdatetime": {' + sLineBreak +
> ' "type": "string",' + sLineBreak +
> ' "description": "Current server time"' + sLineBreak +
> ' }' + sLineBreak +
> ' }' + sLineBreak +
> ' },' + sLineBreak +
> '}';
> 
> maybe these errors are related

Would that be the JsonSchema of the SwaggerDoc example?
![image](https://user-images.githubusercontent.com/13373632/80927170-3bf63200-8d72-11ea-9976-1a350c1cd7b8.png)

Is working as expected

**Comment by st1gok** [comment]
> Would that be the JsonSchema of the SwaggerDoc example?

yes, if this object then working, but me need array objects

**Comment by st1gok** [comment]
> > Hi!
> > I use RC6 and applied changes from fix #376
> > When you use
> > MVCSwagParam(plQuery, 'types', 'types', ptArray, false,'','money,loading,unloading')
> > result in json:
> > {
> > "in": "query",
> > "name": "types",
> > "description": "types",
> > "type": "array",
> > "enum": [
> > "money",
> > "loading",
> > "unloading"
> > ]
> > }
> > but it should be:
> > {
> > "name": "types",
> > "in": "query",
> > "description": "types",
> > "type": "array",
> > **"items": {**
> > **"type": "string",**
> > "enum": [
> > "money",
> > "loading",
> > "unloading"
> > ]
> > }
> > }
> 
> Using the last commit from dmvc I did these tests and when I define the parameter type as an array, the generated JsonSchema shows it as an array. However when you use enumerators the value of an enumerator cannot be an array. See https://swagger.io/docs/specification/2-0/enums/

I tested this in editor.swagger.io

array may contain enum

**Comment by joaoduarte19** [comment]
> yes, if this object then working, but me need array objects

This object represented as an array should look like this:
```json
{
	"type": "array",
	"items": {
		"type": "object",
		"properties": {
			"application": {
				"type": "string",
				"description": "Application Name"
			},
			"online": {
				"type": "boolean",
				"description": "Defines if the server is online"
			},
			"serverdatetime": {
				"type": "string",
				"description": "Current server time"
			}
		}
	}
}
```

**Comment by st1gok** [comment]
> > yes, if this object then working, but me need array objects
> 
> This object represented as an array should look like this:
> 
> ```json
> {
> 	"type": "array",
> 	"items": {
> 		"type": "object",
> 		"properties": {
> 			"application": {
> 				"type": "string",
> 				"description": "Application Name"
> 			},
> 			"online": {
> 				"type": "boolean",
> 				"description": "Defines if the server is online"
> 			},
> 			"serverdatetime": {
> 				"type": "string",
> 				"description": "Current server time"
> 			}
> 		}
> 	}
> }
> ```

I agree see my post https://github.com/danieleteti/delphimvcframework/issues/377#issuecomment-623094609 but this not worked

**Comment by joaoduarte19** [comment]
I think you're missing something. In my tests it is working correctly;
![Untitled-1](https://user-images.githubusercontent.com/13373632/80928101-b7f37880-8d78-11ea-85b7-77e1cd38353d.png)

See this attached sample. Enumerators and array of objects work correctly:
[swaggerdoc.zip](https://github.com/danieleteti/delphimvcframework/files/4572195/swaggerdoc.zip)

**Comment by st1gok** [comment]
Ok, I tested tomorrow and answer

**Comment by st1gok** [comment]
> I think you're missing something. In my tests it is working correctly;
> ![Untitled-1](https://user-images.githubusercontent.com/13373632/80928101-b7f37880-8d78-11ea-85b7-77e1cd38353d.png)
> 
> See this attached sample. Enumerators and array of objects work correctly:
> [swaggerdoc.zip](https://github.com/danieleteti/delphimvcframework/files/4572195/swaggerdoc.zip)

Yes, array of objects work correctly, I have extra comma in my example.
Enums work is single, but not array of enum.

**Comment by joaoduarte19** [comment]
> Enums work is single, but not array of enum.


Indeed, swagger middleware does not support enumerators as an array. We will implement this functionality soon.

**Comment by st1gok** [comment]
Is it possible for now to implement this parameter through AJsonSchema?

**Comment by joaoduarte19** [comment]
I believe it is possible. However it is necessary to test to be sure of this.

**Comment by st1gok** [comment]
I couldn’t do it (

**Comment by blitt001** [comment]
What is the status of this problem ? We are facing similair problem now.

**Comment by joaoduarte19** [comment]
> What is the status of this problem ? We are facing similair problem now.

When we add support for OpenAPI 3.0 we intend to correct this problem.
What problem are you having? You can describe it, so we can try to solve it.
